usr/sbin/p11sak/p11sak.c

```diff
@@ -112,14 +112,16 @@ static void load_pkcs11lib(void)
     atexit(unload_pkcs11lib);
 }
 
-static CK_RV get_pin(char **pin, size_t *pinlen)
+static CK_RV get_pin_prompt(char **pin, size_t *pinlen)
 {
     struct termios old, new;
     int nread;
     char *user_input = NULL;
     size_t buflen = 0;
     CK_RV rc = 0;
 
+    printf("Please enter user PIN: ");
```

Comment:
maybe an fflush(stdout) to ensure it is really printed? As far as I remember printf only flushes when the string contains '\n' ....

Reply:
As far as I can see, the fflush() is not required here, because it happend anyways as soon as the getline() starts (so, right in time...).

Reply:
I don't quite understand.... Does getline flush stdout ? 
I do see a fflush after the getline, but thats too late.

Reply:
It happens implicitely. In a debug session, I get the output in the line with the getline(). But you're right. It is cleaner to do it explicitely. I will att the fflush for the printf here.